teamMaster.py

```
@@ -0,0 +1,264 @@
+        logger.critical(errmsg)
+        logger.exception(e)
+        raise LoadDictionaryError(errmsg)
+
+
+def reset_todays_games_in_master(masterdict):
+
+    keylist = list(masterdict)
+    for key in keylist:
+        keys = list(masterdict[key])
+        if 'today_1' in keys:
+            del masterdict[key]['today_1']
+        if 'today_2' in keys:
+            del masterdict[key]['today_2']
+
+    return masterdict
+
+
+def process_schedule(sched_dict, mstr_dict):
+    """
+    :param sched_dict: daily schedule file used to drive team extract process
+    """
+
+    mstr_dict = reset_todays_games_in_master(mstr_dict)
+
+    logger.info('Applying updates to team master dictionary')
+
+    # get dailySchedule key list and start with blank result team dictionary
+    keylist = list(sched_dict)
+    daily_team_dict = {}
+
+    # for each key, go to boxscore for that game and exctract all team data
+    for key in keylist:
+
+        boxscoreurl = BOXSCORE.replace('/_directory_/',
+                                       sched_dict[key]['directory'])
+        logger.info('Loading boxscore dictionary: ' + boxscoreurl)
+
+        try:
+            boxresp = requests.get(boxscoreurl)
+            boxscore_dict = json.loads(boxresp.text)
+            box_level_3 = boxscore_dict['data']['boxscore']
+        except Exception:
+            errmsg = 'Boxscore dictionary not created yet for: ' + key
+            logger.warning(errmsg)
+            continue
+
+        # set home and away team codess, init team result dict for this game
+        home = sched_dict[key]['home_code']
+        away = sched_dict[key]['away_code']
+
+        # single game or first game of a double header
+        if key[-1:] == '1':
+            try:
+                hist = mstr_dict[home]['game_hist']
+                hist.update({box_level_3['game_id']:
+                             sched_dict[key]['directory']})
+            except Exception:
```

As above be specific about what kind of exception else you'll just eat all the exceptions and when something really goes wrong you won't know it, also I think this deserves a comment to explain what this next line does.

---

There is a comment both if the try works and if the exception is raised...

    try:
        # grab team game history from master and add current game
        hist = mstr_dict[team]['game_hist']
        hist.update({box_level_3['game_id']: game_dir})
    except Exception:
        # first time game history being created for this team
        hist = {box_level_3['game_id']: game_dir}

Does it need to be more detailed? Basically if there is no game history for the team, meaning today is their first game, then establish the 'game_hist' subdictionary within the team entry using the current game.

---

Don't know how I missed this previously.